src/ObjectLock.php

```diff
@@ -88,8 +88,180 @@ public static function getLockableObjects()
+        $showAskUnlock = $useremail->getFromDBByCrit([
+            'users_id'     => $this->fields['users_id'],
+            'is_default'   => 1
+        ]) && ($CFG_GLPI['notifications_mailing'] == 1);
+
+        $userdata = getUserName($this->fields['users_id'], 2);
+
+        if ($showAskUnlock) {
+            $ret = Html::scriptBlock("
+         function askUnlock() {
+            " . Html::jsConfirmCallback(__('Ask for unlock this item?'), $this->itemtypename . " #" . $this->itemid, "function() {
+                  $.post({
+                     url: '" . $CFG_GLPI['root_doc'] . "/ajax/unlockobject.php',
+                     cache: false,
+                     data: {
+                        requestunlock: 1,
+                        id: {$this->fields['id']}
+                     },
+                     dataType: 'json',
+                     success: function( data, textStatus, jqXHR ) {
+                           " . Html::jsAlertCallback($userdata['name'], __('Request sent to')) . "
+                        }
+                     });
+               }") . "
+         }");
+            echo $ret;
+        }
+
+        $ret = Html::scriptBlock("
+         $(function(){
+            var lockStatusTimer;
+            $('#alertMe').on('change',function( eventObject ){
+               if( this.checked ) {
+                  lockStatusTimer = setInterval( function() {
+                     $.get({
+                           url: '" . $CFG_GLPI['root_doc'] . "/ajax/unlockobject.php',
+                           cache: false,
+                           data: 'lockstatus=1&id=" . $this->fields['id'] . "',
+                           success: function( data, textStatus, jqXHR ) {
+                                 if( data == 0 ) {
+                                    clearInterval(lockStatusTimer);" .
+                                    Html::jsConfirmCallback(__('Reload page?'), __('Item unlocked!'), "function() {
+                                       window.location.reload(true);
+                                    }") . "
+                                 }
+                              }
+                           });
+                  },15000)
+               } else {
+                  clearInterval(lockStatusTimer);
+               }
+            });
+         });
+      ");
+
+        $msg = "<strong class='nowrap'>";
+        $msg .= sprintf(__('Locked by %s'), "<a href='" . $user->getLinkURL() . "'>" . $userdata['name'] . "</a>");
+        $msg .= "&nbsp;" . Html::showToolTip($userdata["comment"], ['link' => $userdata['link'], 'display' => false]);
+        $msg .= " -&gt; " . Html::convDateTime($this->fields['date']);
+        $msg .= "</strong>";
+        if ($showAskUnlock) {
+            $msg .= "<a class='btn btn-sm btn-primary ms-2' onclick='javascript:askUnlock();'>
+               <i class='fas fa-unlock'></i>
+               <span>" . __('Ask for unlock') . "</span>
+            </a>";
```

Comment:
Since #16538, most of the front code has been moved in a dedicated JS files and a dedicated Twig template.